src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -151,11 +156,30 @@ public ManagedObjectWrapperHolder(ManagedObjectWrapper* wrapper)
             }
         }
 
-        internal unsafe struct IUnknownVftbl
+        internal unsafe class NativeObjectWrapper
         {
-            public delegate* unmanaged<IntPtr, ref Guid, out IntPtr, int> QueryInterface;
-            public delegate* unmanaged<IntPtr, uint> AddRef;
-            public delegate* unmanaged<IntPtr, uint> Release;
+            private readonly IntPtr _externalComObject;
+            private readonly ComWrappers _comWrappers;
+            public GCHandle _proxyHandle;
```

Comment:
```suggestion
            internal GCHandle _proxyHandle;
```
does not need to be public

Reply:
Curious, what's your rule of thumb on visibility. I usually control visibility mostly using classes and leave all surface of internal classes as public to not bother, until `publicity` happens. Can you share your perspective?

Reply:
I believe that the usual style in runtime libraries is to try to make the visilibity right from the start even for internal types, especially for fields. The hypothetical question to ask - if we were to make this type public for some reason, is the visibility going to be right?

If you search through libraries sources, I think you will find a lot more internal fields in internal types than public fields in internal types.